### PR TITLE
feat(fqdn): giving some time for FQDN resolution

### PR DESCRIPTION
**Pull request by rdefosse**
I missed that one.

### CODE REVIEW

File: src/common/utils/fqdn.cpp [hunk 1]
```diff
@@ -23,36 +23,51 @@
 #include "logger.hpp"
 #include <boost/asio.hpp>
 #include <iostream>
+#include <chrono>
+#include <thread>
+
+#define MAX_NB_RESOLVE_TRIES 4
+#define TIME_BETWEEN_TRIES 2
```
Comment (lionelgo) [inline src/common/utils/fqdn.cpp:30]: I would recommend a time unit indication (second or milliseconds for example)

File: src/common/utils/fqdn.cpp [hunk 2]
```diff
@@ -23,36 +23,51 @@
...
+      for (auto it = endpoints.cbegin(); it != endpoints.cend(); it++) {
+        // get the first Endpoint
+        boost::asio::ip::tcp::endpoint endpoint = *it;
+        address = endpoint.address().to_string();
+        port    = endpoint.port();
+        Logger::spgwu_app().debug(
+            "Resolved a DNS (name %s, protocol %s): Ip Addr %s, port %u",
+            host_name.c_str(), protocol.c_str(), address.c_str(), port);
+        if (endpoint.address().is_v4())
+          addr_type = 0;
```
Comment (lionelgo) [inline src/common/utils/fqdn.cpp:56]: I would recommend the use of kConstants here instead of 0,1, or use boolean isV4, etc

**Review by lionelgo — CHANGES_REQUESTED** [review]

File: src/common/utils/fqdn.cpp (hunk 2 above)
Comment (rdefosse) [reply src/common/utils/fqdn.cpp:56]: I did not touch that part.
This is the original code.

File: src/common/utils/fqdn.cpp (hunk 1 above)
Comment (rdefosse) [reply src/common/utils/fqdn.cpp:30]: `#define TIME_IN_SECS_BETWEEN_TRIES 2` ?

Comment (lionelgo) [reply src/common/utils/fqdn.cpp:30]: yes for example

Comment (lionelgo) [reply src/common/utils/fqdn.cpp:30]: so the reader don't have to search for unit in the code

**Review by lionelgo — APPROVED** [review]

**Pull request merged**